packages/inventory-compliance/src/SystemRulesTable.test.js

```diff
@@ -0,0 +1,157 @@
+                    label: 'CCE-80440-1',
+                    system: 'https://nvd.nist.gov/cce/index.cfm',
+                    __typename: 'RuleIdentifier'
+                },
+                references: [],
+                __typename: 'Rule'
+            }
+        ]
+    }
+];
+
+describe('SystemRulesTable component', () => {
+    it('should render', () => {
+        global.insights = windowInsightsSpy;
+        remediationsApi.getResolutionsBatch.mockImplementation(
+            () => new Promise((callback) => callback(remediationsResponse))
+        );
+        const wrapper = shallow(
+            <SystemRulesTable
+                profileRules={ profileRules }
+                loading={ false }
+                system={ system }
+            />
+        );
+        expect(toJson(wrapper)).toMatchSnapshot();
+    });
+
+    it('should render a loading table', () => {
+        const wrapper = shallow(
+            <SystemRulesTable
+                profileRules={ profileRules }
+                loading={ true }
+                system={ system }
+            />
+        );
+        expect(toJson(wrapper)).toMatchSnapshot();
+    });
+
+    it('should render filtered rows', () => {
+        global.insights = windowInsightsSpy;
+        remediationsApi.getResolutionsBatch.mockImplementation(
+            () => new Promise((callback) => callback(remediationsResponse))
+        );
+        const wrapper = shallow(
+            <SystemRulesTable
+                profileRules={ profileRules }
+                loading={ false }
+                system={ system }
+            />
+        );
+        const instance = wrapper.instance();
+        return instance.setInitialCurrentRows().then(() => {
+            expect(wrapper.state('currentRows').length / 2).toEqual(2);
+            instance.updateFilter(false, [ 'high' ], []).then(() => {
+                expect(wrapper.state('currentRows').length / 2).toEqual(0);
+                instance.updateFilter(false, [ 'high', 'medium' ], []).then(() => {
+                    expect(wrapper.state('currentRows').length / 2).toEqual(1);
+                    instance.handleSearch('docker').then(() => {
+                        expect(wrapper.state('currentRows').length / 2).toEqual(1);
+                    });
+                });
+            });
+        });
+    });
+
+    it('should render sorted rows', () => {
+        global.insights = windowInsightsSpy;
+        remediationsApi.getResolutionsBatch.mockImplementation(
+            () => new Promise((callback) => callback(remediationsResponse))
+        );
+        const wrapper = shallow(
+            <SystemRulesTable
+                profileRules={ profileRules }
+                loading={ false }
+                system={ system }
+            />
+        );
+        const instance = wrapper.instance();
+        return instance.setInitialCurrentRows().then(() => {
+            expect(wrapper.state('currentRows').length / 2).toEqual(2);
```

Comment:
Same as above, refactor this to more streamlined approach.

Reply:
Not sure if this is possible here, the nesting is minimal here - just setting the rows and then sorting.